### PR TITLE
ENH: crosstabs analysis added to autolysis

### PR DESCRIPTION
**Pull request by skathirmani**

### CODE REVIEW

**Comment by skathirmani** [comment]
@pratapvardhan: can we add the module `orderedattrdict` to travis dependency?. The test is failing because of that

**Comment by pratapvardhan** [comment]
Can you add orderedattrdict to requirements.txt, travis will also be taken care of. And, please squash the commits, if you can.

**Comment by pratapvardhan** [comment]
@skathirmani could you also confirm if the docstring for crosstab(s) are upto date?

**Comment by skathirmani** [comment]
@pratapvardhan I have squashed the commits and added `orderedattrdict` to `requirements.txt`. I have cross checked the `crosstabs` doc strings at high level. Anyway I have to add detailed description about the output, which I will be doing later.. Requesting you to review and merge the commit

**Comment by pratapvardhan** [comment]
Failure is due to lack of `orderedattrdict` on conda channel. 

cc @sanand0

File: autolysis/__init__.py [hunk 1]
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
...
+    tells us whether this difference is significant enough to say that
+    ``language`` and ``city`` are independent::
+
+        In [7]: result['V']
+        Out[7]: 0.54866228139712236
+
+    Any value above 0.5 is a strong indication of a relationship. This says
+    that ``language`` and ``city`` are dependent.
+    '''
+    observed = pd.crosstab(index, column, values, aggfunc='sum').replace('', np.nan).fillna(0)
```
Comment (pratapvardhan) [inline autolysis/__init__.py]: line too long, could we multi-line it?

File: autolysis/__init__.py [hunk 2]
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
...
+    pairs_top: integer, Pick only top 10000 pairs by default
+    '''
+    if columns is None:
+        columns = types(data)['groups']
+
+    result = AttrDict()
+    parameters = ('p', 'chi2', 'dof', 'V')
+    for index, column in itertools.combinations(columns, 2):
+        agg_col = values if values in data.columns else column
+        agg_func = bz.count(data[agg_col]) if agg_col == column else bz.sum(data[agg_col])
```
Comment (pratapvardhan) [inline autolysis/__init__.py]: line too long, could we multi-line it?

**Comment by skathirmani** [comment]
@pratapvardhan: It passes flake8

File: autolysis/__init__.py [hunk 3]
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
...
+                               bz.by(bz.merge(data[index], data[column]),
+                                     values=agg_func).head(pairs_top))
+        r = _crosstab(data_grouped[index],
+                      column=data_grouped[column],
+                      values=data_grouped['values'],
+                      correction=correction)
+        result[(index, column)] = {'observed': r['observed'],
+                                   'expected': r['expected'],
+                                   'stats': {param: r[param] for param in parameters}
+                                   }
```
Comment (pratapvardhan) [inline autolysis/__init__.py]: Can be in-line

```python
        result[(index, column)] = {
            'observed': r['observed'],
            'expected': r['expected'],
            'stats': {param: r[param] for param in parameters}
            }
```

File: autolysis/__init__.py [hunk 4]
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
...
+        data = pd.DataFrame({
+            'city':  ['London', 'Shanghai', 'London', 'Shanghai'],
+            'language': ['English', 'English', 'Mandarin', 'Mandarin'],
+            'people': [90, 50, 40, 350],
+        })
+
+        result = autolyse.crosstab(
+            index=data['city'],
+            column=data['language'],
+            values=data['people'])
```
Comment (pratapvardhan) [inline autolysis/__init__.py:270]: should be `autolyse._crosstab`, but would this function be public?

File: autolysis/__init__.py (hunk 2 above)
Comment (pratapvardhan) [inline autolysis/__init__.py]: It's 90 char long, how does it pass, what are you cooking? =)

Comment (skathirmani) [inline autolysis/__init__.py]: @pratapvardhan : Anand has changed the line char limit to 99.. Checkout [here](https://github.com/gramener/autolysis/blob/crosstabs/setup.cfg#L2)

Comment (pratapvardhan) [inline autolysis/__init__.py]: Duh, please ignore all other line-length comments, Apologies. Look into the docstring thing.

**Comment by pratapvardhan** [comment]
Thanks @sanand0 and  @skathirmani . Merging into dev.

**Pull request merged**